### PR TITLE
Spark standalone and HA

### PR DESCRIPTION
**Pull request by ktsakalozos**

### CODE REVIEW

File: lib/charms/layer/apache_spark.py [hunk 1]
```diff
@@ -233,7 +356,11 @@ def start(self):
         # stop services (if they're running) to pick up any config change
         self.stop()
         # always start the history server, start master/worker if we're standalone
-        utils.run_as('ubuntu', '{}/sbin/start-history-server.sh'.format(spark_home), 'hdfs:///user/ubuntu/directory')
+        history_server_event_logs = '/tmp/spark-events'
```
Comment (kwmonroe) [inline lib/charms/layer/apache_spark.py]: Let's use the self.dist_config.path('spark_events') here.  And let's use it in the hdfs case as well, eg:

```python
history_server_event_logs = self.dist_config.path('spark_events')
if (hdfs):
  history_server_event_logs = "hdfs://%s" % self.dist_config.path('spark_events')
```

File: reactive/spark.py [hunk 2]
```diff
@@ -1,70 +1,173 @@
...
-    hookenv.status_set('active', 'Ready')
+    report_status(spark)
 
 
-@when('spark.started')
-@when_not('hadoop.ready')
-def stop_spark():
-    hookenv.status_set('maintenance', 'Stopping Apache Spark')
+@when('spark.started', 'yarn.configured')
+@when_not('hadoop.ready', )
```
Comment (kwmonroe) [inline reactive/spark.py]: Remove extra `,`

File: reactive/spark.py [hunk 3]
```diff
@@ -1,70 +1,173 @@
...
+        spark.configure()
+        spark.start()
+        set_state('zookeeper.configured')
+        report_status(spark)
+
+
+@when('spark.installed', 'zookeeper.configured')
+@when_not('zookeeper.ready')
+def disable_zookeepers():
+    hookenv.status_set('maintenance', 'Disabling high availability in standalone mode')
```
Comment (kwmonroe) [inline reactive/spark.py]: Small knit-pick on the status message.  You could be disabling HA in yarn mode too, right?  I think drop the "in standalone mode" from the status_set.

Comment (ktsakalozos) [inline reactive/spark.py]: Done.

File: reactive/spark.py (hunk 2 above)
Comment (ktsakalozos) [inline reactive/spark.py]: Done

File: lib/charms/layer/apache_spark.py (hunk 1 above)
Comment (ktsakalozos) [inline lib/charms/layer/apache_spark.py]: As we said the events log dir is set to /spark-history/tmp to match what we have for mapred

**Comment by kwmonroe** [comment]
LGTM, +1.  Thanks for addressing those comments!

**Pull request merged**